File: runtime/src/core/private/log/Logger.ts
import { LoggerLevel } from './LoggerLevel'

export interface Logger {
  setLevel(level: LoggerLevel): Promise<void>
  trace(...msg: any[]): Promise<void>
  debug(...msg: any[]): Promise<void>
  info(...msg: any[]): Promise<void>
  warn(...msg: any[]): Promise<void>
  error(...msg: any[]): Promise<void>
}

File: packages/visionOS/web-spatial/Utils/Logger.swift
import Foundation

class Logger {
    static func getLogger() -> Logger {
        return Logger()
    }

    static func initLogger() {}

    func error(_ str: String) {
        print("error: " + str)
    }

    func verbose(_ str: String) {
        print("verbose: " + str)
    }

    func debug(_ str: String) {
        print("debug: " + str)
    }

    func info(_ str: String) {
        print("info: " + str)
    }

    func warning(_ str: String) {
        print("warning: " + str)
    }
}
